pkg/controller/service/servicecontroller.go

```diff
@@ -378,7 +378,9 @@ func (s *ServiceController) createLoadBalancer(service *api.Service) error {
 		return err
 	}
 	name := s.loadBalancerName(service)
-	status, err := s.balancer.EnsureTCPLoadBalancer(name, s.zone.Region, net.ParseIP(service.Spec.LoadBalancerIP),
+	// getPortsForLB already verified that the protocol matches for all ports.
+	// The cloud provider will verify the protocol is supported
```

Comment:
I think it is important to note in this comment that all ports have already had their protocols verified to match, as well as that it is now the responsibility to check that the protocol is supported for that provider.  Before this was not the case, as the servicecontroller enforced TCP only.  How would you like to capture this?